server/routes/api/integrations/schema.ts

```diff
@@ -41,9 +42,16 @@ export const IntegrationsCreateSchema = BaseSchema.extend({
     /** Integration service */
     service: z.nativeEnum(UserCreatableIntegrationService),
 
+    /** Integration access token (or) api key */
+    accessToken: z.string().optional(),
+
+    /** Integration refresh token */
+    refreshToken: z.string().optional(),
+
     /** Integration config/settings */
-    settings: z
-      .object({ url: z.string().url() })
+    settings: MattermostIntegrationSettingsSchema.or(
```

Comment:
Match mattermost settings first so that zod parsing doesn't short circuit to the base `{ url: string }` object.
Better way would be to introduce a type field in every object and use discrimination unions - could be done as a separate effort, imo.

Reply:
Yes we started doing this with newer objects. A hard lesson learned